rfcs/0030-ensure-index-less-than-length-in-since/0030-ensure-index-less-than-length-in-since.md

```diff
@@ -1 +1,87 @@
-In review, see <https://github.com/nervosnetwork/rfcs/pull/223>
+---
+Number: "0030"
+Category: Consensus (Soft Fork)
+Status: Draft
+Author: Ian Yang
+Organization: Nervos Foundation
+Created: 2021-02-04
+---
+
+# Ensure That Index Is Less Than Length In the Input Since Field Using Epoch With Fraction
+
+## Abstract
+
+This document proposes to fix a bug in the CKB reference implementation. When a transaction input uses the epoch with fraction in the `since` field, the `value` must ensure that the index is less than the length.
+
+## Motivation
+
+The `value` part is an encoded number `epoch + index / length`. In CKB, every block is also assigned an epoch number with a fraction. If the block is the `I`th block in the epoch which epoch number is `E` and length is `L`, its epoch number with the fraction is `E + I / L` and `I` must be less than `L`. 
+
+To verify that a transaction is mature since an absolute epoch number with a fraction, the current implementation assumes that `index` must be less than `length` and checks that:
+
+```
+E > epoch
+or (E == epoch and I / L >= index/length)
+```
+
+This is not consistent when using relative flag in the `since` field, where the `value` is considered as a rational number and added to the cell start epoch.
```

Comment:
Another point: up till this point, we have been talking about `value` in the block header part of CKB. If I understand this correctly, the consensus rule for this part is not changed. We are only changed the validation rule for a similar value used in `since` field represented as epoch number. My suggestion is to clarify this fact, clearly stating what is changed.

Right now what is happening here, is that we spend a huge deal of paragraph documenting what is not changed here, and only using a single sentence below this comment talking about what is changed.